parser/indexedfile.go

```diff
@@ -138,7 +139,16 @@ func indexFiles(files []string, indexingThreads int, res *resources.Resources) [
 	}
 
 	indexingWG.Wait()
-	return output
+
+	// remove all nil values from the slice
+	indexedFiles := make([]*fpt.IndexedFile, 0, len(output))
+	for _, file := range output {
+		if file != nil {
+			indexedFiles = append(indexedFiles, file)
+		}
+	}
```

Comment:
This is the piece that fixed 3. The code above pre-allocates a slice in order to make it work with goroutines. But some of the routines would end up leaving nil in the slice. Previously, this slice, nils and all, would get returned and handled elsewhere. Now the nils get stripped out, leaving an empty slice if no valid files were found. This makes it easier to check if there were files we could import or not earlier.